indigo-frontend/src/app/pages/project/project-add/project-add.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, inject, OnInit, ViewChild } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormBuilder, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatInputModule } from '@angular/material/input';
+import { ButtonComponent } from '@/core/components/common/button/button.component';
+import { FileUploadComponent } from '@/core/components/common/file-upload/file-upload.component';
+import { InputComponent } from '@/core/components/common/input/input.component';
+import { ModalComponent } from '@/core/components/common/modal/modal.component';
+import { ApiService } from '@/core/services/api.service';
+import { ProjectRefreshService } from '../project-refresh.service';
+
+@Component({
+  standalone: true,
+  selector: 'app-project-add',
+  imports: [ModalComponent, ButtonComponent, InputComponent, MatInputModule, FileUploadComponent, FormsModule, ReactiveFormsModule, CommonModule],
+  templateUrl: './project-add.component.html'
+})
+export class ProjectAddComponent<T> implements OnInit {
+  @ViewChild('addProjectModal') modalComponent!: ModalComponent;
+  @ViewChild(FileUploadComponent) fileUpload!: FileUploadComponent;
+  files: File[] = [];
+  formGroup!: FormGroup;
+  chips: string[] = [];
+  isSubmitted = false;
+  allowedFileTypes = ['doc'];
+
+  constructor(private service: ApiService<T>, private fb: FormBuilder, private projectRefreshService: ProjectRefreshService){
+    this.service = inject(ApiService);
```

Comment:
Seems you don't need to inject it here as you've already injected it via constructor. So, choose one approach: via inject function or via constructor.